### DISCUSSION TITLE
Grid block for Firestore (or external data source) ?

### DISCUSSION THREAD

**Discussion by MikeYates** [Budibase/budibase#11865] [Help] [answered]
Is there a work around to create a grid block using Firestore as the data source?

I tried playing around with SuperTable as the cells can be edit (if enabled), but it wont let user edit if field is empty.

**Comment by melohagan** [comment, marked as answer]
Hey @MikeYates 

Currently the Grid block only works with data table datasources such as the Budibase Internal DB and SQL databases.

Work is currently being done to support all datasources however!

**Reply by MikeYates** [reply]
@melohagan 

Awesome news, just wondering is there a plan for dynamic components?

**Reply by melohagan** [reply]
Dynamic components, is that creating components from JSON?

**Reply by MikeYates** [reply]
@melohagan 

Yes I believe so, I had to google what that was. Sorry I am no web developer, only an enthusiast.

But yes basically end user can add much number of row on the screen to save, and I dont have to create all fixed texfields having it all visible.

**Reply by melohagan** [reply]
I understand you - it's a popular feature request, but it would be a fairly large chunk of work and there are no upcoming plans for it at the moment.

**Reply by MikeYates** [reply]
Thank you @melohagan I just needed some closure
